### PR TITLE
Teams Tabs Sample [Experimental]

### PR DESCRIPTION
**Pull request by mdrichardson**
Fixes: Our lack of this sample in Node.

Needs testing, since I can't get on a Teams ring that has Tab Fetch/Submit yet. @EricDahlvang?

### CODE REVIEW

File: experimental/teams-tabs/javascript_nodejs/pages/index.html [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+            <h1>Welcome to Teams <em>Tabs</em></h1>
+        </div>
+    </header>
+    <article class="l-article">
+        <p>Generate a Microsoft Teams application.</p>
+    
+        <p>
+            Add the Bot to Microsoft Teams:
+                
+            <a href='https://teams.microsoft.com/l/chat/0/0?users=28:"@Model.MicrosoftAppId"'><img height="30" width="113" src='https://dev.botframework.com/Client/Images/Add-To-MSTeams-Buttons.png'></a>
```
Comment (axelsrz) [inline experimental/teams-tabs/javascript_nodejs/pages/index.html:40]: I believe this is the syntax for templating in .NET, will this work the same in Node?

File: experimental/teams-tabs/javascript_nodejs/bots/teamsAdaptiveCardTabsBot.js [hunk 2]
```diff
@@ -0,0 +1,194 @@
...
+                await context.sendActivity('Hello, I am a Teams Adaptive Card Tabs bot. Please use the tabs to interact. (Send "logout" to sign out)');
+            }
+
+            // By calling next() you ensure that the next BotHandler is run.
+            await next();
+        });
+    };
+
+    async handleTeamsTabFetch(context, tabRequest) {
+        if (tabRequest?.tabEntityContext?.tebEntityId === 'workday') {
```
Comment (axelsrz) [inline experimental/teams-tabs/javascript_nodejs/bots/teamsAdaptiveCardTabsBot.js:36]: typo for tabEntityId ?

File: experimental/teams-tabs/javascript_nodejs/bots/teamsAdaptiveCardTabsBot.js [hunk 3]
```diff
@@ -0,0 +1,194 @@
...
+            let magicCode;
+            if (tabRequest.state) {
+                magicCode = tabRequest.state.toString();
+            }
+
+            return this.getPrimaryTabResponse(context, magicCode);
```
Comment (axelsrz) [inline experimental/teams-tabs/javascript_nodejs/bots/teamsAdaptiveCardTabsBot.js:42]: I believe this should be awaited

File: experimental/teams-tabs/javascript_nodejs/.env [hunk 4]
```diff
@@ -0,0 +1,3 @@
+MicrosoftAppId=
+MicrosoftAppPassword=
+BaseUrl=https://YourDeployedBotUrl.com
```
Comment (axelsrz) [inline experimental/teams-tabs/javascript_nodejs/.env:3]: I think ConnectionName should be added as well

File: experimental/teams-tabs/javascript_nodejs/bots/teamsAdaptiveCardTabsBot.js (hunk 3 above)
Comment (mdrichardson) [reply experimental/teams-tabs/javascript_nodejs/bots/teamsAdaptiveCardTabsBot.js:42]: [JavaScript is just weird](https://eslint.org/docs/rules/no-return-await)

File: experimental/teams-tabs/javascript_nodejs/deploymentTemplates/template-with-new-rg.json [hunk 5]
```diff
@@ -0,0 +1,198 @@
...
+                                    }
+                                }
+                            }
+                        },
+                        {
+                            "apiVersion": "2017-12-01",
+                            "type": "Microsoft.BotService/botServices",
+                            "name": "[parameters('botId')]",
+                            "location": "global",
+                            "kind": "bot",
```
Comment (BruceHaley) [inline experimental/teams-tabs/javascript_nodejs/deploymentTemplates/template-with-new-rg.json:159]: "kind": "bot" is out of date, no longer supported. See my earlier comment.

File: experimental/teams-tabs/javascript_nodejs/deploymentTemplates/template-with-preexisting-rg.json [hunk 6]
```diff
@@ -0,0 +1,167 @@
...
+                    }
+                }
+            }
+        },
+      {
+        "apiVersion": "2017-12-01",
+        "type": "Microsoft.BotService/botServices",
+        "name": "[parameters('botId')]",
+        "location": "global",
+        "kind": "bot",
```
Comment (BruceHaley) [inline experimental/teams-tabs/javascript_nodejs/deploymentTemplates/template-with-preexisting-rg.json:135]: "kind": "bot" is out of date, no longer supported. See my earlier comment.

**Review by BruceHaley — CHANGES_REQUESTED** [review]

**Comment by EricDahlvang** [comment]
Closing, we've decided to not include this.

**Pull request closed without merging**